src/cobalt.rs

```diff
@@ -128,8 +128,10 @@ pub fn build(config: &Config) -> Result<()> {
             !template_extensions.contains(&f.path()
                                             .extension()
                                             .unwrap_or(OsStr::new(""))) &&
-            f.path() != dest && f.path() != layouts_path.as_path()
+            f.path() != dest && !f.path().starts_with(Path::new("./").join(dest)) &&
```

Comment:
Not sure if this is solving the problem cleanly enough. From what I can gather you want to prevent copying things that are inside the /dest folder. I think a better way to do this is to replace the `filter` above with `filter_entry` from walkdir (see http://burntsushi.net/rustdoc/walkdir/#example-skip-hidden-files-and-directories-efficiently-on-unix)